apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -0,0 +1,335 @@
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+        [Fact]
+        public async Task Logging_SizedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<SizedBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+                for (int i = 0; i < 250; i++)
+                {
+                    await client.GetAsync($"/Main/Debug/{testId}");
+                    await client.GetAsync($"/Main/Info/{testId}");
+                    await client.GetAsync($"/Main/Warning/{testId}");
+                    await client.GetAsync($"/Main/Error/{testId}");
+                    await client.GetAsync($"/Main/Critical/{testId}");
+                    await client.GetAsync($"/Main/Exception/{testId}");
+                }
+            }
+
+            // Just check that a large portion of logs entires were pushed.  Not all
+            // will be pushed as some may be in the buffer.
+            var results = GetEntries(startTime, testId, 500);
+            Assert.NotNull(results);
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Debug));
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Info));
+            Assert.Null(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Critical));
+        }
+
+
+        [Fact]
+        public async Task Logging_TimedBuffer()
+        {
+            string testId = Utils.GetTestId();
+            DateTime startTime = DateTime.UtcNow;
+
+            var builder = new WebHostBuilder().UseStartup<TimedBufferLoggerTestApplication>();
+            using (TestServer server = new TestServer(builder))
+            {
+                var client = server.CreateClient();
+
+                await client.GetAsync($"/Main/Warning/{testId}");
+                await client.GetAsync($"/Main/Error/{testId}");
+
+                var noResults = GetEntries(startTime, testId, 0);
+                Assert.Null(noResults);
+                Thread.Sleep(TimeSpan.FromSeconds(70));
+
+                await client.GetAsync($"/Main/Error/{testId}");
+                await client.GetAsync($"/Main/Critical/{testId}");
+            }
+
+            // The fourth entry is not pushed as it is buffered after the last push.
+            var results = GetEntries(startTime, testId, 3);
+            Assert.Equal(3, results.Count());
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Warning));
+            Assert.NotNull(results.FirstOrDefault(l => l.Severity == LogSeverity.Error));
```

Comment:
It might be better to have this verify that there are 2 errors rather than just trying to find the first?